src/main/bash_autocomplete/ads.sh

```diff
@@ -0,0 +1,77 @@
+# This is a work in progress autocompletion script
+# the main skeleton of it is here, but it is missing a few important features
+# most importantly, the auto-complete for services
+
+# To install:
+# ln -s `pwd`/ads.sh /usr/local/etc/bash_completion.d/ads
+
+_ads () {
+    # === BASH Background Reading... ===
+    # COMP_WORDS is the array of arguments right now
+    # COMP_CWORD is the index into that array for the last word
+    # COMPREPLY is a list of all the words that this word could compelte to
+    # ===
+
+    local BASIC_COMMAND_LIST="help list up down status logs bounce home edit"
+    local OPTIONAL_ARGS="-h --help"
+    local FULL_COMMAND_LIST="$OPTIONAL_ARGS $BASIC_COMMAND_LIST"
+
+    # wordlist will be a space delimited list of words that they could complete to
+    wordlist="UNHANDLED"    
+
+    # curw is the word they are currently typing
+    local curw=${COMP_WORDS[COMP_CWORD]}
+
+    # The first argument to ads is a command
+    if [ "$COMP_CWORD" -eq "1" ]
```

Comment:
bash style in this project is to put then on the same line. we don't want to have to rewrite our whole repo's history in the future =)